lid_mismatches.py

```diff
@@ -0,0 +1,166 @@
+    )
+    parser.add_argument(
+        "--wrong_language_csv",
+        type=str,
+        default="lid_wrong_language_samples.csv",
+        help="Filename for the output CSV with samples where both LIDs are wrong.",
+    )
+    parser.add_argument(
+        "--mismatch_samples_csv",
+        type=str,
+        default="lid_mismatch_samples.csv",
+        help="Filename for the output CSV with samples where either the input or target LID differs from the assigned language.",
+    )
+
+    args = parser.parse_args()
+
+    # Load the dataset
+    dataset = load_dataset(args.dataset_name)
+    df = pd.DataFrame(dataset[args.dataset_split])
+
+    # Load the tokenized CSV file
+    tokenized_df = pd.read_csv(args.tokenized_csv_filename)
+
+    # Load the lid CSV file
+    lid_df = pd.read_csv(args.lid_csv_filename)
+
+    # Merge dataset with LID results
+    m_df = pd.concat([df, lid_df, tokenized_df], axis=1)
+
+    # 1. Drop duplicates
+    duplicated_df = m_df[m_df.duplicated(subset=["inputs", "targets"], keep="first")]
+    duplicated_df.to_csv(args.duplicated_samples_csv, index=False)
+    print(f"Duplicated samples saved to {args.duplicated_samples_csv}")
+    m_df.drop_duplicates(subset=["inputs", "targets"], keep="first", inplace=True)
+
+    # 2. Filter out rows based on minimum length criteria
+    short_df = m_df[
+        (m_df["inputs_n_tokens_stanza"] < 4)
+        & (m_df["targets_n_tokens_stanza"] < 4)
+        & (df["inputs"].apply(lambda s: not any([s.strip().endswith(c) for c in "?؟"])))
+    ]
+    short_df.to_csv(args.short_samples_csv, index=False)
+    print(f"Short samples saved to {args.short_samples_csv}")
+    m_df.drop(short_df.index, inplace=True)
+
+    # Filter out rows based on minimum length criteria
+    merged_df = m_df[
+        (m_df["inputs"].str.len() >= args.min_inputs_length)
+        & (m_df["targets"].str.len() >= args.min_targets_length)
+    ]
+
+    # 1. Generate clean samples where LIDs match the assigned language
+    clean_samples_df = merged_df[
+        (merged_df["language_code"] == merged_df[args.inputs_lid_col])
+        & (merged_df["language_code"] == merged_df[args.targets_lid_col])
+    ]
+    clean_samples_df.to_csv(args.clean_samples_csv, index=False)
+    print(f"Clean samples saved to {args.clean_samples_csv}")
+
+    # 2. Generate samples where both LIDs are wrong
+    wrong_language_df = merged_df[
+        (merged_df["language_code"] != merged_df[args.inputs_lid_col])
+        & (merged_df["language_code"] != merged_df[args.targets_lid_col])
+    ]
+    wrong_language_df.to_csv(args.wrong_language_csv, index=False)
+    print(f"Samples with both LIDs wrong saved to {args.wrong_language_csv}")
+
+    # 3. Generate samples with mismatched LIDs (either input or target differs from assigned language)
+    mismatch_samples_df = merged_df[
+        (merged_df["language_code"] != merged_df[args.inputs_lid_col])
+        ^ (merged_df["language_code"] != merged_df[args.targets_lid_col])
```

Comment:
@AMR-KELEG just to be clear, we want XOR not OR in here? If so, the comment on 137 should probably get updated

Reply:
Yes, as the ones with both inputs and targets lids that are different were already flagged in the previous step. So, no need to have them in two different splits.